crypto/keys/private.go

```diff
@@ -23,8 +24,16 @@ import (
 	"errors"
 	"fmt"
 	"io/ioutil"
+
+	"github.com/google/trillian"
 )
 
+// Provider handles acquisition of signers for trees.
+type Provider interface {
```

Comment:
`Generate` is also a factory method in a sense, since it creates new objects of a type that the provider will decide.  What do you think? 

Btw, having interfaces with optional implementations seems less than ideal.  If `Generate` will be optional, maybe there's a slightly different place to put it? Perhaps a utility method that's not part of the interface?